### PR TITLE
Show error message when final result doesn't have default metric

### PR DESCRIPTION
**Pull request by Lijiaoa**
image

### CODE REVIEW

File: src/webui/src/App.tsx [hunk 1]
```diff
@@ -33,8 +39,34 @@ class App extends React.Component<{}, AppState> {
...
         this.setState({ metricGraphMode: (EXPERIMENT.optimizeMode === 'minimize' ? 'min' : 'max') });
+        // final result is legal
+        // get a succeed trial，see final result data's format
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.dataFormatimer = window.setInterval(this.getFinalDataFormat, this.state.interval * 1000);
     }
 
+    getFinalDataFormat = (): void => {
+        for(let i = 0; this.state.isilLegalFinal === false; i++){
+            if(TRIALS.succeededTrials()[0] !== undefined && TRIALS.succeededTrials()[0].final !== undefined){
```
Comment (ultmaster) [inline src/webui/src/App.tsx]: What if we don't have succeeded trials?

File: src/webui/src/App.tsx [hunk 2]
```diff
@@ -33,8 +39,34 @@ class App extends React.Component<{}, AppState> {
...
         this.timerId = window.setTimeout(this.refresh, this.state.interval * 1000);
         this.setState({ metricGraphMode: (EXPERIMENT.optimizeMode === 'minimize' ? 'min' : 'max') });
+        // final result is legal
+        // get a succeed trial，see final result data's format
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.dataFormatimer = window.setInterval(this.getFinalDataFormat, this.state.interval * 1000);
     }
 
+    getFinalDataFormat = (): void => {
+        for(let i = 0; this.state.isilLegalFinal === false; i++){
```
Comment (ultmaster) [inline src/webui/src/App.tsx]: ilLegal? Really? Should be illegal.

File: src/webui/src/App.tsx [hunk 3]
```diff
@@ -54,30 +86,34 @@ class App extends React.Component<{}, AppState> {
...
         const reactPropsChildren = React.Children.map(this.props.children, child =>
             React.cloneElement(
                 child as React.ReactElement<any>, {
-                    interval,
-                    columnList, changeColumn: this.changeColumn,
-                    experimentUpdateBroadcast,
-                    trialsUpdateBroadcast,
-                    metricGraphMode, changeMetricGraphMode: this.changeMetricGraphMode
+                interval,
+                columnList, changeColumn: this.changeColumn,
```
Comment (ultmaster) [inline src/webui/src/App.tsx]: Don't change indentation.

Comment (Lijiaoa) [reply src/webui/src/App.tsx]: ok

File: src/webui/src/App.tsx (hunk 1 above)
Comment (Lijiaoa) [reply src/webui/src/App.tsx]: line 64, quit for loop.

**Comment by Lijiaoa** [comment]
fix issue #1724 #2087

File: src/webui/src/App.tsx (hunk 1 above)
Comment (ultmaster) [reply src/webui/src/App.tsx]: I mean `TRIALS.succeddedTrials()` will return an empty array, right? What happens to `x[0]` if `x` is an empty array?

**Pull request merged**